docs/source/1.0/spec/core/model.rst

```diff
@@ -0,0 +1,2705 @@
+==============
+
+The values provided for :ref:`members <member>` are either always present
+and set to a default value when necessary or *boxed*, meaning a value is
+optionally present with no default value. Members are considered boxed if
+the member is marked with the :ref:`box-trait` or the shape targeted by the
+member is marked with the box trait. Members that target strings, timestamps,
+and aggregate shapes are always considered boxed and have no default values.
+
+- The default value of a ``byte``, ``short``, ``integer``, ``long``,
+  ``float``, and ``double`` shape that is not boxed is zero.
+- The default value of a ``boolean`` shape that is not boxed is ``false``.
+- All other shapes are always considered boxed and have no default value.
+
+
+Recursive shape definitions
+===========================
+
+Smithy allows for recursive shape definitions with the following constraint:
+the member of a list, set, or map cannot directly or transitively target its
+containing shape unless one or more members in the path from the container
+back to itself targets a structure or union shape. This ensures that shapes
+that are typically impossible to define in various programming languages are
+not defined in Smithy models (for example, you can't define a recursive list
+in Java ``List<List<List....``).
+
+The following recursive shape definition is **valid**:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        list ValidList {
+            member: IntermediateStructure
+        }
+
+        structure IntermediateStructure {
+            foo: ValidList
+        }
+
+    .. code-tab:: json
+
+        {
+            "smithy": "1.0",
+            "shapes": {
+                "smithy.example#ValidList": {
+                    "type": "list",
+                    "member": {
+                        "target": "smithy.example#IntermediateStructure"
+                    }
+                },
+                "smithy.example#IntermediateStructure": {
+                    "type": "structure",
+                    "members": {
+                        "foo": {
+                            "target": "smithy.example#ValidList"
+                        }
+                    }
+                }
+            }
+        }
+
+The following recursive shape definition is **invalid**:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        namespace smithy.example
+
+        list RecursiveList {
+            member: RecursiveList
+        }
+
+    .. code-tab:: json
+
+        {
+            "smithy": "1.0",
+            "shapes": {
+                "smithy.example#RecursiveList": {
+                    "type": "list",
+                    "member": {
+                        "target": "smithy.example#RecursiveList"
+                    }
+                }
+            }
+        }
+
+
+.. _service-types:
+
+--------------
+Service shapes
+--------------
+
+*Service types* have specific semantics and define services, resources,
```

Comment:
IMO, service coming first makes sense because it's described as the "starting point". It's the most abstract thing you need to know before drilling into its components.